qiskit_nature/circuit/library/ansatzes/evolved_operator_ansatz.py

```diff
@@ -44,6 +45,11 @@ def __init__(
             name: The name of the circuit.
             initial_state: A `QuantumCircuit` object to prepend to the circuit.
         """
+        warn_deprecated_same_type_name(
+            "0.2.0",
+            DeprecatedType.CLASS,
+            "qiskit.circuit.library.EvolvedOperatorAnsatz",
+        )
```

Comment:
This works a little bit differently. Right now, the following `DeprecationWarning` will be raised by your code:
```python
<stdin>:1: DeprecationWarning: The qiskit.circuit.library.EvolvedOperatorAnsatz class is deprecated as of version 0.2.0 and will be removed no sooner than 3 months after the release. Instead use the qiskit.circuit.library.EvolvedOperatorAnsatz class.
```

Thus, you need to change the above to the following:
```suggestion
warn_deprecated_same_type_name(
            "0.2.0",
            DeprecatedType.CLASS,
            "EvolvedOperatorAnsatz",
            "from qiskit.circuit.library as a direct replacement",
        )
```
This will then produce the expected message:
```python
<stdin>:1: DeprecationWarning: The EvolvedOperatorAnsatz class is deprecated as of version 0.2.0 and will be removed no sooner than 3 months after the release. Instead use the EvolvedOperatorAnsatz class from qiskit.circuit.library as a direct replacement.
```

Reply:
Thank you very much. your warning message must be better.